### ISSUE TITLE
When used with an array, the array order is not maintained.

### ISSUE THREAD

**Issue by sbuys** [d4rekanguok/gatsby-schema-field-absolute-path#8]

**Comment by d4rekanguok** [comment]
Thanks for catching that, the plugin is querying the data with $in so the result is not sorted.

https://github.com/d4rekanguok/gatsby-schema-field-absolute-path/blob/d3ccbac1edf857f83245b9356e515bf56232f0e2/src/gatsby-node.ts#L48-L67

To fix the issue we could sort the result array based on `filePaths`, which maintains the order of the original field's array.

```js
const sorted = filePaths.map(
  absPath => fileNodes.find(({ absolutePath }) => absolutePath === absPath )
)
```

Will test it out when I can! If you have time & interested, PRs are very welcomed. Thank you!

**Comment by sbuys** [comment]
Was thinking the same thing 👍 . Testing it now. Just dealing with some typescript plumbing issues on my box.  🤓

**Comment by d4rekanguok** [comment]
Fixed #9
